From 4a2afd26e23784580d901fa67bb4f6bede4eccc5 Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Wed, 7 Oct 2009 03:32:02 -0500
Subject: [PATCH] dpa: Don't initialize disabled devices

Device nodes might have a "status" property, declaring the device as
"disabled", and if so, we should not initialize the device.

Signed-off-by: Andy Fleming <afleming@freescale.com>
[Kevin: Original patch taken from Freescale p4080 SDK 2.0 ISO image.
Apply cleanly to kernel 2.6.27]
Integrated-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/dpa/dpa.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/net/dpa/dpa.c b/drivers/net/dpa/dpa.c
index d02e39e..469552d 100644
--- a/drivers/net/dpa/dpa.c
+++ b/drivers/net/dpa/dpa.c
@@ -2082,6 +2082,9 @@ dpa_init_probe(struct of_device *_of_dev)
 
 	dpa_node = _of_dev->node;
 
+	if (!of_device_is_available(dpa_node))
+		return -ENODEV;
+
 	/* FM */
 
 	mac_dev = dpa_mac_probe(_of_dev);
@@ -2230,6 +2233,7 @@ dpa_probe(struct of_device *_of_dev)
 	const uint32_t		*rx_fqids;
 	int			num_tx_fqids, num_tx_fqs;
 	int			num_rx_fqids, num_rx_fqs;
+	const char *dpa_status;
 
 	dev = &_of_dev->dev;
 
@@ -2237,6 +2241,9 @@ dpa_probe(struct of_device *_of_dev)
 
 	dpa_node = _of_dev->node;
 
+	if (!of_device_is_available(dpa_node))
+		return -ENODEV;
+
 	/*
 	 * Allocate this early, so we can store relevant information in
 	 * the private area
-- 
1.6.0.4

